linker-interface/shared/src/main/scala/org/scalajs/linker/interface/ModuleOutputSpec.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.linker.interface
+
+/** Module output specification. */
+final class ModuleOutputSpec private (modules: Seq[(ModuleOutputSpec.Selector, ModuleOutputSpec.Module)]) {
```

Comment:
We need to finely balance what is accessible in this interface. The use case of a build tool would require access to the list of modules to prepare the relevant output files.

However, `deps` and `initializers` should probably not be accessible. (if we are to make `deps` accessible, we should make the verified and potentially inferred deps accessible).